### ISSUE TITLE
MICCAI pytorch implementation

### ISSUE THREAD

**Issue by SteffenCzolbe** [voxelmorph/voxelmorph#148]
Hello Adrian,

I would like to use the voxel morph registration architecture in my preferred library PyTorch. You published an initial version of a PyTorch implementation of the CVPR net, with the vague hint that you might publish more in the future. Is a pytorch version of the net presented at MICCAI in the works?

If not, I will start coding the MICCAI net in pytorch. Do you accept pull requests? If so, I can contribute the code.

**Comment by adalca** [comment]
Hi @SteffenCzolbe 

Thanks for your message.

The MICCAI model had essentially two practical improvements: the first is the diffeomorphisms achieved in practice by scaling and squaring, and the second is the probabilistic nature. We've implemented the scaling and squaring in pytorch in the `redesign` branch, which is our current effort to refactor and organize voxelmorph which I'm hoping we'll be able to release soon (but is a branch you can look at). We have not implemented the probabilistic aspect. If you'd like to help with that, we would certainly consider pull requests -- but just keep in mind that the redesign branch is what we'll be using soon.

Does this make sense?

**Comment by yellowgardenia** [comment]
Hi, I am using your code to run the brain registration task with default lambda and sigma and an increased number of integrating layers,  there would be large number of irregular deformations when running the MICCAI version, not as mentioned in your paper. I found other papers also reported the same issue, like "Learning a Probabilistic Model for Diffeomorphic Registration". So can you explain what may cause this issue?  Thanks.

**Comment by adalca** [comment]
hi @yellowgardenia ,

Could you specify a bit more which code you are running (e.g. keras/pytorch) and with what data? Is the data normalized to [0, 1] for example? And by irregular do you mean regions where |J|<0 ?

**Comment by yellowgardenia** [comment]
We run the Keras with the LONI data. The data has been normalized to [0,1] before training. And the irregular refers to the regions |J|<0.

**Comment by MaddyWyburd** [comment]
Hello @adalca ,
I am interested in using your scaling and squaring MICCAI implementation in Pytorch. I see from above this has been implemented in the 'redesign' branch, however I cannot seem to access this branch. Has it been removed/closed? 
Many thanks, Maddy

**Comment by adalca** [comment]
Hi @MaddyWyburd ,

This should be in the `master` branch now, in the `torch` folder

**Comment by MaddyWyburd** [comment]
Hello @adalca. Amazing, many thanks
